src/interpreter/mod.rs

```diff
@@ -383,9 +383,12 @@ pub(crate) fn parse_closure(
                 .map(|n| n == "codeBody")
                 .and_then(|b| b.then_some(token))
                 .unwrap_or_else(|| &token.children[0]);
-            let arg_count = code_body_uses_lambda_parameter(code_body) as usize;
-            let mut new_scope =
-                scope.child_scope(ArgCount::Fixed(arg_count), engine.create_return_target());
+            let arg_count = if code_body_uses_lambda_parameter(code_body) {
+                ArgCount::Fixed(1)
+            } else {
+                ArgCount::Range { min: 0, max: 1 }
```

Comment:
The only inconsistency this creates is that closures can implicitly take and ignore one parameter if they don't specify it. This makes sense to me since these kinds of closures are used in contexts where they will have either 0 or 1 argument, and it would be tedious to need to check if the function takes an argument in every place where it's called. This might be worth discussing more at the team meeting.